Doppler.AccountPlans/Utils/CalculateUpgradeCostHelper.cs

```diff
@@ -7,19 +7,35 @@ public static class CalculateUpgradeCostHelper
     {
         public static PlanAmountDetails CalculatePlanAmountDetails(PlanInformation newPlan, PlanDiscountInformation newDiscount, PlanInformation currentPlan, DateTime now)
         {
-            var currentBaseMonth = now.Day >= 21 ? currentPlan.CurrentMonthPlan : currentPlan.CurrentMonthPlan - 1;
-            var totalPlan = newDiscount.MonthPlan != currentBaseMonth ? (newPlan.Fee - currentPlan.Fee) * (1 - (newDiscount.DiscountPlanFee / 100)) : newPlan.Fee * (1 - (newDiscount.DiscountPlanFee / 100));
+            var isMonthPlan = currentPlan.CurrentMonthPlan != 0;
 
-            return new PlanAmountDetails
+            var currentMonthPlan = isMonthPlan ?
+                currentPlan.CurrentMonthPlan :
+                1;
+
+            var currentBaseMonth = now.Day < 21 ?
+                currentMonthPlan - 1 :
+                currentMonthPlan;
+
+            var differenceBetweenMonthPlans = newDiscount.MonthPlan - currentBaseMonth;
+
+            var discountMonthPlans = (isMonthPlan || differenceBetweenMonthPlans == 0) ?
```

Comment:
`discountMonthPlans `should be `differenceBetweenMonthPlans `, right?

Reply:
What do you think about numberOfMonthsWithDiscount?

Reply:
Yes numberOfMonthsToDiscount